src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -5624,12 +5638,17 @@ private static String replace(final String text, String searchString, final Stri
              if (--max == 0) {
                  break;
              }
-             end = searchText.indexOf(searchString, start);
+             end = indexOf(text, searchString, start, ignoreCase);
          }
          buf.append(text, start, text.length());
          return buf.toString();
      }
 
+     private static int indexOf(String str, String searchStr, int fromIndex, boolean ignoreCase) {
```

Comment:
I will move this method below indexOf(CharSequence, CharSequence, int) declaration around line 1473

Reply:
java doc suggestions are welcome